components/Wallet/Send.tsx

```diff
@@ -0,0 +1,251 @@
+    resetWalletError()
+    const newMessage = new Message({
+      to: message.to,
+      from: message.from,
+      nonce: await walletProvider.getNonce(wallet.address),
+      value: message.value,
+      method: message.method,
+      params: message.params,
+      gasPremium: gasParams.gasPremium.toAttoFil(),
+      gasFeeCap: gasParams.gasFeeCap.toAttoFil(),
+      gasLimit: new BigNumber(gasParams.gasLimit.toAttoFil()).toNumber()
+    })
+    try {
+      const lotusMessage = newMessage.toLotusType()
+      const signedMessage = await walletProvider.wallet.sign(
+        wallet.address,
+        lotusMessage
+      )
+      const msgValid = await walletProvider.simulateMessage(lotusMessage)
+      if (!msgValid) {
+        throw new Error('Filecoin message invalid. No gas or fees were spent.')
+      }
+      await walletProvider.sendMessage(signedMessage)
+      navigate(router, { pageUrl: PAGE.WALLET_HOME })
+    } catch (e: any) {
+      logger.error(e)
+      setSendError(e)
+    }
+    setIsSending(false)
+  }
+
+  return (
+    <Dialog>
+      <StandardBox>
+        <h2>Send Filecoin</h2>
+        <hr />
+        <p>Please enter the message details below</p>
+      </StandardBox>
+      {gasParamsError && (
+        <ErrorBox>
+          Failed to calculate gas fees: {gasParamsError.message}
+        </ErrorBox>
+      )}
+      {sendError && (
+        <ErrorBox>Failed to send message: {sendError.message}</ErrorBox>
+      )}
+      {walletError() && (
+        <ErrorBox>The wallet produced an error: {walletError()}</ErrorBox>
+      )}
+      {isSending && (
+        <Transaction.Confirm loginOption={loginOption as LoginOption} />
+      )}
+      <ShadowBox>
+        <Transaction.Header address={wallet.address} balance={wallet.balance} />
+        <form>
+          <InputV2.Address
+            label='Recipient'
+            autofocus={true}
+            value={toAddress}
+            onBlur={setMessageIfChanged}
+            onChange={setToAddress}
+            setIsValid={setIsToAddressValid}
+            disabled={gasParamsLoading || isSending}
+          />
+          <InputV2.Filecoin
+            label='Amount'
+            max={wallet.balance}
+            value={value}
+            denom='fil'
+            onBlur={setMessageIfChanged}
+            onChange={setValue}
+            setIsValid={setIsValueValid}
+            disabled={gasParamsLoading || isSending}
+          />
+          <InputV2.Params
+            label='Params'
+            value={params}
+            onBlur={setMessageIfChanged}
+            onChange={setParams}
+            setIsValid={setIsParamsValid}
+            disabled={
+              loginOption === LoginOption.LEDGER ||
+              gasParamsLoading ||
+              isSending
+            }
+            info={
+              loginOption === LoginOption.LEDGER
+                ? 'Ledger devices cannot sign base64 params yet, coming soon'
+                : ''
+            }
```

Comment:
I have to double check this - ledger may now be able to sign arbitrary params

Reply:
Cool, let me know!